package/MDAnalysis/analysis/density.py

```diff
@@ -408,8 +416,20 @@ def _prepare(self):
             # ideal solution would use images: implement 'looking across the
             # periodic boundaries' but that gets complicated when the box
             # rotates due to RMS fitting.
-            smin = np.min(coord, axis=0) - self._padding
-            smax = np.max(coord, axis=0) + self._padding
+            try:
+                smin = np.min(coord, axis=0) 
+                smax = np.max(coord, axis=0) 
+            except ValueError as err:
+                msg = ("No atoms in selection at current frame, running user grid")
+                warnings.warn(msg)
+                logger.warning(msg)
+                smin = self._gridcenter 
```

Comment:
Can you detail the logic here? This shouldn't work since you can't get here if `self._gridcenter` is not `None`?

Reply:
Ooh, perhaps it should fail here; Since no user defined grid is present in this case!